src/database/models/categories.rs

```diff
@@ -453,3 +465,297 @@ impl<'a> GameVersionBuilder<'a> {
         Ok(GameVersionId(result.id))
     }
 }
+
+#[derive(Default)]
+pub struct LicenseBuilder<'a> {
+    pub short: Option<&'a str>,
+    pub name: Option<&'a str>,
+}
+
+impl License {
+    pub fn builder() -> LicenseBuilder<'static> {
+        LicenseBuilder::default()
+    }
+
+    pub async fn get_id<'a, E>(id: &str, exec: E) -> Result<Option<LicenseId>, DatabaseError>
+    where
+        E: sqlx::Executor<'a, Database = sqlx::Postgres>,
+    {
+        let result = sqlx::query!(
+            "
+            SELECT id FROM licenses
+            WHERE short = $1
+            ",
+            id
+        )
+        .fetch_optional(exec)
+        .await?;
+
+        Ok(result.map(|r| LicenseId(r.id)))
+    }
+
+    pub async fn get<'a, E>(id: LicenseId, exec: E) -> Result<License, DatabaseError>
+    where
+        E: sqlx::Executor<'a, Database = sqlx::Postgres>,
+    {
+        let result = sqlx::query!(
+            "
+            SELECT short, name FROM licenses
+            WHERE id = $1
+            ",
+            id as LicenseId
+        )
+        .fetch_one(exec)
+        .await?;
+
+        Ok(License {
+            id,
+            short: result.short,
+            name: result.name,
+        })
+    }
+
+    pub async fn list<'a, E>(exec: E) -> Result<Vec<License>, DatabaseError>
+    where
+        E: sqlx::Executor<'a, Database = sqlx::Postgres>,
+    {
+        let result = sqlx::query!(
+            "
+            SELECT id, short, name FROM licenses
+            "
+        )
+        .fetch_many(exec)
+        .try_filter_map(|e| async {
+            Ok(e.right().map(|c| License {
+                id: LicenseId(c.id),
+                short: c.short,
+                name: c.name,
+            }))
+        })
+        .try_collect::<Vec<License>>()
+        .await?;
+
+        Ok(result)
+    }
+
+    pub async fn remove<'a, E>(short: &str, exec: E) -> Result<Option<()>, DatabaseError>
+    where
+        E: sqlx::Executor<'a, Database = sqlx::Postgres>,
+    {
+        use sqlx::Done;
+
+        let result = sqlx::query!(
+            "
+            DELETE FROM licenses
+            WHERE short = $1
+            ",
+            short
+        )
+        .execute(exec)
+        .await?;
+
+        if result.rows_affected() == 0 {
+            // Nothing was deleted
+            Ok(None)
+        } else {
+            Ok(Some(()))
+        }
+    }
+}
+
+impl<'a> LicenseBuilder<'a> {
+    /// The game version.  Spaces must be replaced with '_' for it to be valid
```

Comment:
Copy-pasted docs, go through and fix the other copies of this - also in the insert method.